File: app/src/main/java/org/deiverbum/app/model/MassReading.java
package org.deiverbum.app.model;

import static org.deiverbum.app.utils.Utils.LS;
import static org.deiverbum.app.utils.Utils.LS2;
import static org.deiverbum.app.utils.Utils.normalizeEnd;

import android.text.SpannableStringBuilder;

import org.deiverbum.app.utils.Utils;

public class MassReading extends Biblical implements Comparable<MassReading>{
    private String tema;

    public String getTema() {
        return tema;
    }

    public String getTemaForRead() {
        return normalizeEnd(tema);
    }

    public void setTema(String tema) {
        this.tema = tema;
    }

    /**
     * <p>Obtiene la lectura bíblica completa, incluyendo el responsorio, formateada para la vista.</p>
     * @since 2022.01
     * @return Un objeto {@link SpannableStringBuilder con el contenido.}
     */
    public SpannableStringBuilder getAll() {
        SpannableStringBuilder sb = new SpannableStringBuilder();
        sb.append(LS);
        sb.append(getHeader());
        sb.append(LS2);
        sb.append(book.getLiturgyName());
        sb.append("    ");
        sb.append(Utils.toRed(getCita()));
        sb.append(LS2);
        if(tema!=null) {
            sb.append(Utils.toRed(getTema()));
            sb.append(LS2);
        }
        sb.append(getTextoSpan());
        sb.append(Utils.LS2);
        return sb;
    }

    /**
     * <p>Obtiene la lectura bíblica completa formateada para la lectura de voz.</p>
     * @since 2022.01
     * @return Un objeto {@link SpannableStringBuilder con el contenido.}
     */
    @Override
    public SpannableStringBuilder getAllForRead(){
        SpannableStringBuilder sb=new SpannableStringBuilder();
        sb.append(Utils.normalizeEnd(findOrden(getOrden())));
        sb.append(getLibroForRead());
        sb.append(getTemaForRead());
        sb.append(getTextoForRead());
        return sb;
    }
    @Override
    public SpannableStringBuilder getHeader() {

        String header="";
        if (this.order >= 1 && this.order <= 19) {
            header="PRIMERA LECTURA";
        }

        if (this.order >= 20 && this.order <= 29) {
            header="SALMO RESPONSORIAL";
        }

        if (this.order >= 30 && this.order <= 39) {
            header="SEGUNDA LECTURA";
        }
        if (this.order >= 40 ) {
            header="EVANGELIO";
        }

        return Utils.formatTitle(header);
    }

    public String findOrden(int orden) {
        String orderText;
        if (orden <= 19) {
            orderText = "Primera Lectura";
        } else if (orden <= 29) {
            orderText = "Salmo Responsorial";
        } else if (orden <= 39) {
            orderText = "Segunda Lectura";
        } else {
            orderText = "Evangelio";
        }
        return orderText;
    }

    @Override
    public int compareTo(MassReading e) {
        return this.getOrden().compareTo(e.getOrden());
    }
}

File: app/src/main/java/org/deiverbum/app/model/MassReading.kt
package org.deiverbum.app.model

import android.text.SpannableStringBuilder
import org.deiverbum.app.util.Utils

open class MassReading : Biblical(), Comparable<MassReading> {
    var tema: String? = null
    private val temaForRead: String
        get() = Utils.normalizeEnd(tema)

    /**
     *
     * Obtiene la lectura bíblica completa, incluyendo el responsorio, formateada para la vista.
     *
     * @return Un objeto [con el contenido.][SpannableStringBuilder]
     * @since 2022.01
     */
    fun getAll(type: Int): SpannableStringBuilder {
        val sb = SpannableStringBuilder()
        sb.append(Utils.LS)
        sb.append(Utils.formatTitle(getHeader(type)))
        sb.append(Utils.LS2)
        sb.append(book?.liturgyName)
        sb.append("    ")
        sb.append(Utils.toRed(quote))
        sb.append(Utils.LS2)
        if (tema != null) {
            sb.append(Utils.toRed(tema))
            sb.append(Utils.LS2)
        }
        sb.append(textoSpan)
        sb.append(Utils.LS2)
        return sb
    }

    /**
     *
     * Obtiene la lectura bíblica completa formateada para la lectura de voz.
     *
     * @return Un objeto [con el contenido.][SpannableStringBuilder]
     * @since 2022.01
     */
    //@Override
    fun getAllForRead(type: Int): SpannableStringBuilder {
        val sb = SpannableStringBuilder()
        sb.append(Utils.normalizeEnd(getHeader(type)))
        sb.append(book?.getForRead())
        sb.append(temaForRead)
        sb.append(textoForRead)
        return sb
    }

    private fun getHeader(type: Int): String {
        return if (type == 0) {
            var header = ""
            if (order!! >= 1 && order!! <= 19) {
                header = "PRIMERA LECTURA"
            }
            if (order!! >= 20 && order!! <= 29) {
                header = "SALMO RESPONSORIAL"
            }
            if (order!! >= 30 && order!! <= 39) {
                header = "SEGUNDA LECTURA"
            }
            if (order!! >= 40) {
                header = "EVANGELIO"
            }
            header
        } else {
            getHeaderByType(type)
        }
    }

    /**
     *
     * Obtiene el encabezado de cada lectura según el tipo.
     *
     * @return Un objeto [con el contenido.][String]
     * @since 2023.1.2
     */
    private fun getHeaderByType(type: Int): String {
        /*
            type 1 es la Vigilia Pascual, con el siguiente esquema:
            1. 1ª Lectura
            2. Salmo
            3. O bien: Salmo
            4. 1ª Lectura
            5. Salmo
            6. 3ª Lectura
            7. Salmo
            8. 4ª Lectura
            9. Salmo
            10. 5ª Lectura
            11. Salmo
            12. 6ª Lectura
            13. Salmo
            14. 7ª Lectura
            15. Salmo
            16. o bien: Salmo
            17. Epístola
            18. Salmo
            40. Evangelio
         */
        var header = ""
        if (type == 1) {
            if (order == 1) {
                header = "PRIMERA LECTURA"
            }
            if (order == 2 || order == 5 || order == 7 || order == 9 || order == 11 || order == 13 || order == 15 || order == 18) {
                header = "SALMO RESPONSORIAL"
            }
            if (order == 3 || order == 16) {
                header = "O bien: SALMO RESPONSORIAL"
            }
            if (order == 4) {
                header = "SEGUNDA LECTURA"
            }
            if (order == 6) {
                header = "TERCERA LECTURA"
            }
            if (order == 8) {
                header = "CUARTA LECTURA"
            }
            if (order == 10) {
                header = "QUINTA LECTURA"
            }
            if (order == 12) {
                header = "SEXTA LECTURA"
            }
            if (order == 14) {
                header = "SÉPTIMA LECTURA"
            }
            if (order == 17) {
                header = "EPÍSTOLA"
            }
            if (order!! >= 40) {
                header = "EVANGELIO"
            }
        }
        return header
    }

    override fun compareTo(other: MassReading): Int {
        return getOrden()!!.compareTo(other.getOrden()!!)
    }
}